vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -39,6 +39,34 @@ def _calculate_option_height(full_options: OptionsType) -> int:
     return 8 + 24 * number_of_lines
 
 
+def _add_select_all_option(full_options: OptionsType, multi: bool):
+    """Adds div component for select all option."""
+    if not multi:
+        return full_options
+
+    def create_select_all_option():
+        """Creates the "Select All" option as a dictionary."""
+        return {
+            "label": html.Div(["ALL"]),
+            "value": "ALL",
+        }
+
+    altered_options = []
+    for option in full_options:
+        if isinstance(option, str):
+            if option == "ALL":
+                altered_options.append(create_select_all_option())
+            else:
+                altered_options.append({"label": option, "value": option})
+        elif isinstance(option, dict):
+            if option.get("value") == "ALL":
+                altered_options.append(create_select_all_option())
+            else:
+                altered_options.append(option)
+
+    return altered_options
```

Comment:
This is quite similar to the code we wrote for the Navigations, where we also had to handle both dictionary and string cases. I really liked our approach back then of first converting everything to a uniform type before applying the logic. I think it made it overall look a bit cleaner - WDYT? :)

Please check below:

```suggestion
def _convert_to_dict(option: Union[str, dict[str, any]]) -> dict[str, any]:
    """Converts a string option to a dictionary format."""
    if isinstance(option, str):
        return {"label": option, "value": option}
    return option

def _add_select_all_option(full_options: OptionsType) -> OptionsType:
    """Adds a 'Select All' option to the list of options."""
    options_dict = [_convert_to_dict(option) for option in full_options]
    select_all_option = {"label": html.Div(["ALL"]), "value": "ALL"}
    return [select_all_option if option.get("value") == "ALL" else option for option in options_dict]
```

Reply:
This suggestion is pretty much similar to what I added here -> https://github.com/mckinsey/vizro/pull/961#discussion_r1923356781

Reply:
I'll take this approach as it does not require changing tests for other selectors that use `get_options_and_default`, but we can refactor in the next iteration when adding functionality if this is preferred approach.